src/gsi-ncdiag/gsi_ncdiag.py

```diff
@@ -185,6 +185,10 @@
     'Nonlinear_QC_Rel_Wgt': 'GsiQCWeight',
     'Errinv_Adjust': 'GsiAdjustObsError',
     'Errinv_Final': 'GsiFinalObsError',
+    'Errinv_Input': 'GsiInputObsError',
+    'SWCM_spec_type': 'SpectralType',
```

Comment:
Do we want to include the BUFR mnemonics in the netCDF variable names?

Reply:
I use them b/c they are unique to the satellite winds and don't leave room for confusion what this variable is. I could of course use more descriptive variable names instead, i.e. SatWind_Comp_Method for SWCM, Sat_Zen_Angle for SAZA, SatChan_Freq for SCCF.

Reply:
I think longer, more descriptive names are desired but I'll defer to Emily/Steve.

Reply:
Taking a look in the output file, it seems that the piece of code here is defining group names (the @ObsValue part of the variable name) for output variables. The data values however get repeated for eastward_wind and northward_wind variables (shown below) with the same group name.
```
        float eastward_wind@SpectralType(nlocs) ;
        float eastward_wind@SatZenAngle(nlocs) ;
        float eastward_wind@ChanWavelen(nlocs) ;

        float northward_wind@SpectralType(nlocs) ;
        float northward_wind@SatZenAngle(nlocs) ;
        float northward_wind@ChanWavelen(nlocs) ;
```

For example, ``eastward_wind@SpectralType(nlocs) = northward_wind@SpectralType(nlocs)``, etc.

To me (and I'm not the expert on these data), these look like variables that maybe belong in the ``@MetaData`` group. For example:

| current variable | new variable |
|-----------------|-------------|
| eastward_wind@SpectralType(nlocs) northward_wind@SpectralType(nlocs) | satwind_comp_spectral_type@MetaData |
| eastward_wind@SatZenAngle(nlocs) northward_wind@SatZenAngle(nlocs) | sensor_zenith_angle@MetaData |
| eastward_wind@ChanWavelen(nlocs) northward_wind@ChanWavelen(nlocs) | channel_wavelength@MetatData |

The above is just a suggestion for the new names. Feel free to edit. The idea is that these variables are meta data describing each location (like latitude@MetaData and longitude@MetaData) instead of observed values.

Is this perhaps the case with these variables?

Reply:
Yes, the new variables data are repeated for eastward_wind and northward_wind, because I was mimicking what was already in the *obs* file (not the smartest move on my end).

A wind observation will always require a pair of eastward_wind@ObsValue and northward_wind@ObsValue. Everything else , eastward_wind@* and northward_wind@* , is repetitive information, i.e. eastward_wind@SpectralType and northward_wind@SpectralType contain the same information, and so on.

If we want to avoid the repetitions (and reduce file size) we can make all of these MetaData.
Does that sound reasonable? (I would have to add them to all_LocKeyList , is that right?)

Reply:
I think that is right. @CoryMartin-NOAA is more familiar with the code and should be able to give a definitive answer. Also, Cory is my speculation about what is going on correct?

Reply:
Yes @srherbener I agree these should be metadata, they are going to be the same at every location for both u,v components of the wind.

Reply:
So, should I change only the new variables to metadata, or also all that are repetitive for the eastward_wind and northward_ wind components?  

And the variables names - keep or change (in ProdGSI) ?

Reply:
What you want to end up with in the output ioda netcdf is the replacement of the "current variables" with the "new variables" in the table above.

I'm not expert with the code/process/etc so @CoryMartin-NOAA could you help @ilianagenkova figure out what to modify in the scripts to get this result? Thanks!

Reply:
@ilianagenkova you can keep the things the same in ProdGSI, I'd just change anything that would be constant at each location to be metadata. Really the only things that should be duplicated would be the ob error, qc and value (or other things of that nature I'm forgetting).

There's a dictionary that contains the metadata that you can add this to and it will only add it to the output file if the input files have the matching keys. Let me know if you need more specific help.

Reply:
@ilianagenkova In gsi_ncdiag.py, look for "all_LocKeyList".  You will see a list of names for MetaData.  That's the place you want to add your new variables that belong to @MetaData.   In the list of "all_LocKeyList",  you will see two names for each entry.  The first name is the one defined in GSI, and the second one is for IODA obs file (the name of your variable in the output IODA obs file).